src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/DirectoryAssemblyResolver.cs

```diff
@@ -114,9 +127,22 @@ public virtual AssemblyDefinition Load (string fileName)
 
 		protected virtual AssemblyDefinition ReadAssembly (string file)
 		{
+			bool haveDebugSymbols = loadDebugSymbols &&
+				(File.Exists (Path.ChangeExtension (file, ".mdb")) ||
+				 File.Exists (Path.ChangeExtension (file, ".pdb")) ||
+				 File.Exists (Path.ChangeExtension (file, ".ppdb")));
 			var reader_parameters = new ReaderParameters () {
-				AssemblyResolver  = this,
-				ReadSymbols       = loadDebugSymbols && (File.Exists(file + ".mdb") || File.Exists(Path.ChangeExtension(file, ".pdb"))),
+				ApplyWindowsRuntimeProjections  = loadReaderParameters.ApplyWindowsRuntimeProjections,
+				AssemblyResolver                = this,
+				MetadataImporterProvider        = loadReaderParameters.MetadataImporterProvider,
+				InMemory                        = loadReaderParameters.InMemory,
+				MetadataResolver                = loadReaderParameters.MetadataResolver,
+				ReadingMode                     = loadReaderParameters.ReadingMode,
+				ReadSymbols                     = haveDebugSymbols,
```

Comment:
I'm sorry, but I don't understand the comment; what does the "this" refer to in "without this being fixed"?

My *guess* is that "this" refers to "`ReaderParameters.ReadSymbols` always set to `true`".

Assuming that's the case, I'm still somewhat dubious of that. The current expectation -- if annoying -- is that when `ReaderParameters.ReadSymbols` is `true` and the symbol file doesn't exist, a `FileNotFoundException` is thrown. Changing that semantic would likely annoy anybody who actually wants to ensure that symbol files exist.

There are thus two solutions I can think of:

1. Always set `ReaderParameters.ReadSymbols` to `true` *anyway*, then catch the exception. (Just because I don't like it doesn't mean it's not the better solution.)
2. Add a new `ReaderParameters.ReadSymbolsWhenPossible` property (or something), which would cause Cecil to try to read symbols if they exist, but not fail if they don't.